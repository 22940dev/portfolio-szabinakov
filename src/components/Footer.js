import React from 'react'

import '../styles/Footer.css'

import twitterlogo from '../pictures/twitter.png'
import instagramlogo from '../pictures/insta.png'
import linkedinlogo from '../pictures/linkedin.png'
import github from '../pictures/github.png'
import gmail from '../pictures/gmail.png'

const Footer = () => {
    return(
        <footer>
            <ul className='footerContainer' >
                <li data-testid='footerTestOne' className='footerListElement'>
                    <a href='https://twitter.com/szabinkov?lang=en-gb' target='blank'>
                    <img src={twitterlogo} href='www.google.com' alt='twitter'/>
                    </a>
                </li>
                <li data-testid='footerTestTwo' className='footerListElement'>
                    <a href='https://www.instagram.com/szabinush/' target='blank'>
                    <img src={instagramlogo} alt='instagram'/>
                    </a>
                </li>
                <li data-testid='footerTestThree' className='footerListElement'>
                    <a href='https://www.linkedin.com/in/szabina-kov%C3%A1cs-7a0592159/' target='blank'>
                    <img src={linkedinlogo} alt='linkedin'/>
                    </a>
                </li>
                <li data-testid='footerTestFour' className='footerListElement'>
                    <a href='https://github.com/szabinakov' target='blank'>
                    <img src={github} alt='Github'/>
                    </a>
                </li>
                <li data-testid='footerTestFive' className='footerListElement'>
                    <a href='mailto:szabinkov@gmail.com'>
                    <img src={gmail} alt='Gmail'/>
                    </a>
                </li>
            </ul>
            </footer>
    )
}

export default Footer