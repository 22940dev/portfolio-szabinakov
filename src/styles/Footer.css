footer {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  left: 0;
  bottom: 0;
}

.footerContainer {
  list-style-type: none;
  width: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
}
.footerContainer > a {
  padding: 10px;
}

.footerContainer > a:hover {
  transform: scale(2, 2);
}
a > img {
  width: 20px;
  border: none;
}

.footerListElement {
  padding-right: 10px;
}
