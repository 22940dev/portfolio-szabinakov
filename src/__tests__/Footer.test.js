import React from 'react'
import {render} from '@testing-library/react'
import Footer from '../components/Footer'


describe('Footer', () => {
    it('matches the Snapshot', () => {
        const {asFragment} = render (<Footer/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders all the links', () => {
        const {getByTestId} = render (<Footer/>)

        expect(getByTestId('footerContainer')).toHaveClass('footerContainer')
        expect(getByTestId('footer')).toHaveClass('footer')
        expect(getByTestId('footerLinkOne')).toHaveClass('footerLinkOne')
        expect(getByTestId('footerLinkTwo')).toHaveClass('footerLinkTwo')
        expect(getByTestId('footerLinkThree')).toHaveClass('footerLinkThree')
        expect(getByTestId('footerLinkFour')).toHaveClass('footerLinkFour')
        expect(getByTestId('footerLinkFive')).toHaveClass('footerLinkFive')
    })
})