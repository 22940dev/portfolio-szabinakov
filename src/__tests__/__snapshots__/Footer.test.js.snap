// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches the Snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
    data-testid="footer"
  >
    <div
      class="footerContainer"
      data-testid="footerContainer"
    >
      <a
        class="footerLinkOne"
        data-testid="footerLinkOne"
        href="https://twitter.com/szabinkov?lang=en-gb"
        target="blank"
      >
        <img
          alt="twitter"
          href="www.google.com"
          src="twitter.png"
        />
      </a>
      <a
        class="footerLinkTwo"
        data-testid="footerLinkTwo"
        href="https://www.instagram.com/szabinush/"
        target="blank"
      >
        <img
          alt="instagram"
          src="insta.png"
        />
      </a>
      <a
        class="footerLinkThree"
        data-testid="footerLinkThree"
        href="https://www.linkedin.com/in/szabina-kov%C3%A1cs-7a0592159/"
        target="blank"
      >
        <img
          alt="linkedin"
          src="linkedin.png"
        />
      </a>
      <a
        class="footerLinkFour"
        data-testid="footerLinkFour"
        href="https://github.com/szabinakov"
        target="blank"
      >
        <img
          alt="Github"
          src="github.png"
        />
      </a>
      <a
        class="footerLinkFive"
        data-testid="footerLinkFive"
        href="mailto:szabinkov@gmail.com"
      >
        <img
          alt="Gmail"
          src="gmail.png"
        />
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
